pkg/resource/api.go

```diff
@@ -60,16 +60,17 @@ func NewAPIManagedCreator(c client.Client, t runtime.ObjectTyper) *APIManagedCre
 func (a *APIManagedCreator) Create(ctx context.Context, cm Claim, cs NonPortableClass, mg Managed) error {
 	cmr := meta.ReferenceTo(cm, MustGetKind(cm, a.typer))
 	csr := meta.ReferenceTo(cs, MustGetKind(cs, a.typer))
-	mgr := meta.ReferenceTo(mg, MustGetKind(mg, a.typer))
 
 	mg.SetClaimReference(cmr)
 	mg.SetNonPortableClassReference(csr)
 	if err := a.client.Create(ctx, mg); err != nil {
 		return errors.Wrap(err, errCreateManaged)
 	}
 
-	meta.AddFinalizer(cm, claimFinalizerName)
+	// Final name/namespace values of the managed resource are determined _after_ creation.
```

Comment:
It might be helpful to explain why this is the case (because we now use generatename, I assume). It wasn't immediately obvious to me.